src/components/Card.js

```diff
@@ -15,44 +39,75 @@ class Card {
     return cardElement;
   }
 
+  toggleLike() {
+    console.log("Toggling Like Button:", this._likeButton);
+    if (!this._likeButton) {
+      console.error("Like Button is undefined in toggleLike!");
+      return;
+    }
+    this._likeButton.classList.toggle("card__like-button_active");
+  }
+
   _setEventListeners() {
     this._likeButton = this._element.querySelector(".card__like-button");
     this._cardImage = this._element.querySelector(".card__image");
     this._trashButton = this._element.querySelector(".card__delete-button");
 
-    if (this._likeButton) {
-      this._likeButton.addEventListener("click", () => {
-        this._handleLikeButton();
-      });
-    }
-
     if (this._cardImage) {
       this._cardImage.addEventListener("click", () => {
         this._handleImageClick(this._link, this._name);
       });
     }
 
+    if (this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    }
+
+    if (this._likeButton) {
+      this._likeButton.addEventListener("click", () => {
+        this._handleLikeCard(this);
+      });
+    }
+
     if (this._trashButton) {
       this._trashButton.addEventListener("click", () => {
-        this._handleDeleteButton();
+        const cardId = this.getID();
+        console.log("Trash Button: Opening confirm popup for ID:", cardId);
+        console.log("Trash Button: Card element:", this._element);
+        this._confirmPopup.open(this._id, this._element);
       });
     }
   }
 
-  _handleLikeButton() {
-    this._likeButton.classList.toggle("card__like-button_active");
-  }
+  _handleLikeCard() {
```

Comment:
Please, delete `_handleLikeCard` because it's reassigned in the constructor. You don't need it here